tests/python_functional/src/common/file.py

```diff
@@ -22,33 +22,84 @@
 #############################################################################
 import logging
 
+from pathlib2 import Path
+
+from src.common.blocking import DEFAULT_TIMEOUT
 from src.common.blocking import wait_until_true
+from src.common.blocking import wait_until_true_custom
 from src.common.operations import open_file
 
 logger = logging.getLogger(__name__)
 
 
 class File(object):
     def __init__(self, file_path):
-        self.__file_path = file_path
+        self.path = Path(file_path)
         self.__opened_file = None
 
     def __del__(self):
-        if self.__opened_file:
-            self.__opened_file.close()
-            self.__opened_file = None
-
-    def __is_file_exist(self):
-        return self.__file_path.exists()
+        if self.is_opened():
+            self.close()
 
     def wait_for_creation(self):
-        file_created = wait_until_true(self.__is_file_exist)
+        file_created = wait_until_true(self.path.exists)
         if file_created:
-            logger.debug("File has been created:\n{}".format(self.__file_path))
+            logger.debug("File has been created: {}".format(self.path))
         else:
-            logger.debug("File not created:\n{}".format(self.__file_path))
+            raise Exception("File was not created in time: {}".format(self.path))
         return file_created
 
     def open(self, mode):
-        self.__opened_file = open_file(self.__file_path, mode)
+        self.__opened_file = open_file(self.path, mode)
```

Comment:
Can we move `open_file` here from `common.operations`? It could be a local helper function.
As I see, `File` class usage is cleared up in Light: everyone uses the `File` class and there's no other user of `open_file`

Reply:
Sorry, if I misunderstood you suggest that to move: `common.operations::open_file()` to `common.file::open_file()` as a local function? If this is the idea, I am ok with that, but I would move all the other general file related functions also such: copy_file(), copy_shared_file(), ...

What do you think? Do you agreed with that?

Reply:
Yes please, that's what I tried to say. :)
Of course, other functions can remain in `common.operations`.

Reply:
I have fixed this.